src/com/amazon/ion/benchmark/ParseAndCompareBenchmarkResults.java

```diff
@@ -57,110 +38,110 @@ public class ParseAndCompareBenchmarkResults {
+        return processedData;
     }
 
     /**
-     * Parse the benchmark result and extract the IonStruct which contains scores information.
-     *
-     * @param benchmarkResultFilePath is the file path of benchmark result.
-     * @param keyWord                 represents which aspect of scores are required to be extracted.
-     * @return an IonStruct which contains the scores information.
-     * @throws Exception if errors occur when create IonReader.
+     * Using Interquartile range method filtering out the outliers to make sure the raw data collection is normally distributed.
+     * @param data represents the raw data from the same benchmarking process but different iterations.
+     * @return the pre-processed data array without outlier.
      */
-    private static IonStruct parseScore(String benchmarkResultFilePath, String keyWord) throws Exception {
-        IonStruct scoreStruct;
-        IonStruct benchmarkResultStruct = readHelper(benchmarkResultFilePath);
-        if (keyWord.equals(SPEED)) {
-            scoreStruct = (IonStruct) benchmarkResultStruct.get(PRIMARY_METRIC);
-        } else {
-            IonStruct secondaryMetricStruct = (IonStruct) benchmarkResultStruct.get(SECONDARY_METRIC);
-            scoreStruct = (IonStruct) secondaryMetricStruct.get(keyWord);
+    public static double[] removeOutliers(double[] data) {
+        Percentile percentile = new Percentile();
+        double q1 = percentile.evaluate(data, 25.0);
+        double q3 = percentile.evaluate(data, 75.0);
+        double iqr = q3 - q1;
+
+        double lowerBound = q1 - 1.5 * iqr;
+        double upperBound = q3 + 1.5 * iqr;
+
+        List<Double> filteredData = new ArrayList<>();
+        for (double value : data) {
+            if (value >= lowerBound && value <= upperBound) {
+                filteredData.add(value);
+            }
         }
-        return scoreStruct;
+        return filteredData.stream().mapToDouble(d -> d).toArray();
     }
 
     /**
-     * Extract the parameter of specific aspect from set (input | options) out of benchmark result.
-     *
-     * @param benchmarkResultFilePath is the path of benchmark result.
-     * @param keyWord                 specifies which parameter will be returned, from set (input | options).
-     * @return an Ion Value represents the parameter conforms with the keyword.
-     * @throws Exception if error occurs when reading Ion data.
+     * Converting IonList to DoubleStream.
+     * @param data represents the input IonList.
+     * @return the converted DoubleStream.
      */
-    private static IonValue getParameter(String benchmarkResultFilePath, String keyWord) throws Exception {
-        IonStruct benchmarkResultStruct = readHelper(benchmarkResultFilePath);
-        IonStruct parameterStruct = (IonStruct) benchmarkResultStruct.get(PARAMETERS);
-        return parameterStruct.get(keyWord);
+    public static DoubleStream toDouble(IonList data){
```

Comment:
This can be private. You should also just return `List<Double>` or `double[]` unless you're going to be passing around `DoubleStream` to other functions. (E.g. if the input and output of `removeOutliers` was `DoubleStream`.)

Reply:
Thanks, will change to private in the next commit. 
The raw data is a nested `IonList`, and `toDouble` is used for transform the inner `IonList` to `DoubleStream` while flattening the raw data. The return value of `toDouble` is not directly passing to `removeOutliers`. [Here](https://github.com/amazon-ion/ion-java-benchmark-cli/pull/58/commits/2b02d3a1cd79441eff09db1aa3a2ba4c23cc460e#diff-53b4790715ee277df76aa3587d5ea213712796275b4394455ea8883b508b7aafR108-R110) is where the data flattening happened. In this case, should we still consider the returned data type as `List<Double>` or `double[]` ?